### PR TITLE
Process: Uncomment disabled tests and make compatible with Darwin

### PR DESCRIPTION
**Pull request by spevans**
- Handle differences in output from /bin/cat on Linux/macOS when
  printing an error.

-  test_pipe_stdout_and_stderr_same_pipe(): Dont pass malloc debug
  flags to subprocesses when debugging under Xcode.

- runTask(): return stdout and stderr as separate values so the malloc debug is captured separately.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci test

File: TestFoundation/TestProcess.swift [hunk 1]
```diff
@@ -206,7 +207,12 @@ class TestProcess : XCTestCase {
             XCTFail("Could not read stdout")
             return
         }
-        XCTAssertEqual(string, "/bin/cat: invalid_file_name: No such file or directory\n")
+
+        // Remove the leading '/bin/' since on macOS '/bin/cat' just outputs 'cat:'
+        let searchStr = "/bin/"
+        let errMsg = string.replacingOccurrences(of: searchStr, with: "", options: [.literal, .anchored],
+                                              range: searchStr.startIndex..<searchStr.endIndex)
+        XCTAssertEqual(errMsg, "cat: invalid_file_name: No such file or directory\n")
```
Comment (alblue) [inline TestFoundation/TestProcess.swift]: This smells a bit fragile to me - we're expecting a specific formatting of the error message and text. Why not just assert that the string contains "No such file or directory" instead?

**Comment by alblue** [comment]
This looks good to me (with only one minor concern regarding exact error messages). @phausler can you approve the review when you get a chance?

File: TestFoundation/TestProcess.swift (hunk 1 above)
Comment (spevans) [reply TestFoundation/TestProcess.swift]: Reasonable point, however I would probably wait until it breaks on a specific platform before loosening it. Also, that would also match if the executable didn't exist:

```
$ /bin/foo
-bash: /bin/foo: No such file or directory
```

Comment (alblue) [reply TestFoundation/TestProcess.swift]: Fair enough :)

**Pull request merged**